compiler/rustc_borrowck/src/diagnostics/region_errors.rs

```diff
@@ -256,6 +263,70 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
         outlives_suggestion.add_suggestion(self);
     }
 
+    fn get_impl_ident_and_self_ty_from_trait(
```

Comment:
It feels to me like we might already have code doing this somewhere 🤔

Reply:
We do already have that functionality in lexical region diagnostics (`NiceRegionError`), but can't directly use that struct in NLL (and at some point we will get rid of it anyway afaict).